### DISCUSSION TITLE
Using sa_type causes Mypy type error

### DISCUSSION THREAD

**Discussion by vinodkumars** [fastapi/sqlmodel#955] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from sqlmodel import SQLModel, Field
from sqlalchemy import DateTime

class BaseModel(SQLModel):
  created_at:Optional[datetime]=Field(sa_type=DateTime(timezone=True))
```

### Description

Creating a column with sa_type causes a type error with Mypy. Here is the error:
```
No overload variant of "Field" matches argument type "DateTime"
```

### Operating System

macOS

### SQLModel Version

0.0.18

### Python Version

3.11.8

**Comment by redb0** [comment]
I also faced the same problem. Are there any solutions?
 
I use `Python 3.9.19` and `sqlmodel 0.0.19`

**Comment by dolfandringa** [comment]
Same basic thing with pyright:
```
Argument of type "DateTime" cannot be assigned to parameter "sa_type" of type "Type[Any] | PydanticUndefinedType" in function "Field"
```

**Comment by monchin** [comment]
It seems that you should use 
```python
from typing import Optional

from datetime import datetime
from sqlmodel import Column, DateTime, Field, SQLModel, func


class DateTimeTest(SQLModel):
    __tablename__ = "date_time_test"
    created_at: Optional[datetime] = Field(
        sa_column=Column(DateTime(timezone=True), server_default=func.now())
    )
```

**Reply by monchin** [reply]
You can see more details [here](https://github.com/fastapi/sqlmodel/issues/370#issuecomment-1169665476)
